pkg/adapters/grpc/secret_grpc.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2021 Nitric Pty Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package grpc
+
+import (
+	"context"
+
+	pb "github.com/nitric-dev/membrane/interfaces/nitric/v1"
+	"github.com/nitric-dev/membrane/pkg/plugins/secret"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// GRPC Interface for registered Nitric Secret Plugins
+type SecretServer struct {
+	pb.UnimplementedSecretServiceServer
+	secretPlugin secret.SecretService
+}
+
+func (s *SecretServer) checkPluginRegistered() error {
```

Comment:
Note I removed service plugin registered check with other plugins as recommended.

Reply:
That one snuck past me, when was this recommended? If the plugins are nil in these cases it will cause the membrane to crash entirely, as the constructor methods aren't providing safety here either. If we're going to remove these checks I would prefer the safety be implemented in constructor methods and RPC server registration skipped an runtime in the membrane.

Reply:
It was something you recommended during a documents PR.  Simple to put these back.

Reply:
I remember recommending removing the `bool` return on the method (originally it returned ok like a cast), but not the entire method.
https://github.com/nitrictech/membrane/pull/76#discussion_r665080267